docs/content/en/docs/design/config.md

```diff
@@ -25,6 +21,43 @@ read the configuration file from the current directory.
 
 You can [learn more]({{< relref "/docs/references/yaml" >}}) about the syntax of `skaffold.yaml`.
 
+Skaffold normally expects to find the configuration file as
+`skaffold.yaml` in the current directory, but the location can be
+overridden with the `--filename` flag.
+
+### File resolution
+
+The Skaffold configuration file often references other files and
+directories.  These files and directories are usually referenced
+relative to the current directory _and not to the location of
+the Skaffold configuration file_.  There is one important exception:
+files referenced from a build artifact definition are resolved
+relative to the build artifact's _context_ directory.
+
+In the following config file, the `Dockerfile` for building `app`
+is resolved relative to the `frontend` directory (i.e., `frontend/Dockerfile`),
+whereas the the Helm chart's location and its values-files are
+relative to the current directory in `helm/project`.
+```yaml
+apiVersion: skaffold/v2beta11
+kind: Config
+build:
+  artifacts:
+  - image: app
+    context: frontend
```

Comment:
should we mention that if `context` is not explicitly defined then it's set to the CWD anyway?